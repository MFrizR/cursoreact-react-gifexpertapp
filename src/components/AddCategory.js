import React, { useState } from 'react';
import PropsTypes from 'prop-types';

const AddCategory = ({ setCategories } ) => {

    const [inputValue, setInputValue] = useState(' ');

    const handleInputChange = (e) => {
        console.log(e.target.value);
        setInputValue (e.target.value)
    }

    const handleSubmit = (e) => {
        // Prevenir comportamiento por defecto del formulario
        e.preventDefault();
        

        if(inputValue.trim().length > 2){
            setCategories( cats => [  inputValue,...cats]);
            setInputValue('');
        }
        console.log('Submit Hecho');
    }

    return (
        <>
        <form onSubmit = { handleSubmit }>
            <input
                type= "text"
                value= { inputValue }
                onChange = { handleInputChange }
            />
        </form>
          

          
        </>
    )
}

AddCategory.propsTypes = {
    setCategories: PropsTypes.func.isRequired
};

 AddCategory.defaultProps = {

    //setCategories: 
};  

export default AddCategory
