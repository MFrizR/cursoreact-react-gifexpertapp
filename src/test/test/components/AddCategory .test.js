import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from "enzyme"
import AddCategory from "../../../components/AddCategory"

describe('Pruebas en <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={ setCategories }/>);

    beforeEach( () => {

        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setCategories }/>);
    })

    test('1.- Debe de mostrarse correctamente ', () => {
        
        expect( wrapper ).toMatchSnapshot();
    })

    test('2.- Debe de cambiar la caja de texto', () => {
        
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change', {target: { value }});

        expect( wrapper.find('p').text().trim()).toBe( value );
    })
    
    test('3.- No debe de postear la informaciÃ³n con submit  ', () => {

        wrapper.find('form').simulate('submit',{ preventDefault(){} })
        //Compara si no se ha llamado. 
        expect( setCategories).not.toHaveBeenCalled();
    })
    
    test('4.- Deben de llamar el setCategories y limpiar la caja de texto ', () => {
        
       const value = 'Hola mundo'; 
       // 1.- Simular el inputChange
        wrapper.find('input').simulate('change', {target: { value }});

        //2.- Simular el submit
        wrapper.find('form').simulate('submit',{ preventDefault(){} });

        // 3.- setCategories se debe haber llamado
        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledTimes(1);
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function));

        //4.- El valor del input debe de estar ''
        expect( wrapper.find( 'input' ).prop('value')).toBe('');

    })
    
})
