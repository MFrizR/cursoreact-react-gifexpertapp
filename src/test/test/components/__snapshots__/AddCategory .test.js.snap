// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> 1.- Debe de mostrarse correctamente  1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <p>
       
    </p>
    <input
      onChange={[Function]}
      type="text"
      value=" "
    />
  </form>
</Fragment>
`;
